### PR TITLE
Add assertion and tests for non unique diffIdentifiers

### PR DESCRIPTION
**Pull request by viktorgardart**
## Changes in this pull request

Issue fixed: #814 

### Checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [x] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: Tests/IGListAdapterTests.m [hunk 1]
```diff
@@ -1248,4 +1248,9 @@ - (void)test_whenSwappingCollectionViews_withMultipleAdapters_thatDoesntNilOther
     XCTAssertEqual(collectionView1.dataSource, adapter2);
 }
 
+- (void)test_whenPassingNonUniqueIdentifiers_adapterShouldAssert {
+    self.dataSource.objects = @[@0, @1, @2, @0];
+    XCTAssertThrows([self.adapter reloadDataWithCompletion:nil]);
```
Comment (rnystrom) [inline Tests/IGListAdapterTests.m]: Great!

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,11 @@ The changelog for `IGListKit`. Also see the [releases](https://github.com/instag
 3.1.0 (**upcoming release**)
 -----
 
-3.0.0 
+### Enhancements
+
+- Added an assertion if the `diffIdentifiers` aren't unique when performing updates. [viktorgardart](https://github.com/viktorgardart) [(#821)](https://github.com/Instagram/IGListKit/pull/821)
```
Comment (rnystrom) [inline CHANGELOG.md]: +1 I don't think we need this, will strip when importing

File: Source/IGListAdapter.m [hunk 3]
```diff
@@ -494,9 +494,15 @@ - (void)updateObjects:(NSArray *)objects dataSource:(id<IGListAdapterDataSource>
     IGParameterAssert(dataSource != nil);
 
 #if DEBUG
+
+    NSCountedSet *identifiersSet = [[NSCountedSet alloc] init];
```
Comment (rnystrom) [inline Source/IGListAdapter.m]: nit: `[NSCountedSet new]` instead

File: Source/IGListAdapter.m [hunk 4]
```diff
@@ -494,15 +494,12 @@ - (void)updateObjects:(NSArray *)objects dataSource:(id<IGListAdapterDataSource>
     IGParameterAssert(dataSource != nil);
 
 #if DEBUG
-
-    NSCountedSet *identifiersSet = [[NSCountedSet alloc] init];
-
+    NSCountedSet *identifiersSet = [NSCountedSet new];
```
Comment (viktorgardart) [inline Source/IGListAdapter.m]: 👍

**Pull request closed without merging**